framework/packages/abstract-std/src/objects/ownership/README.md

```diff
@@ -92,11 +96,87 @@ pub fn query(deps: Deps, env: Env, msg: QueryMsg) -> StdResult<Binary> {
 }
 ```
 
-## Edge cases
+### Edge cases
+
+## NFT governance type
+
+Utility for querying the owner of a specific NFT. In case NFT contract does not return owner of `owner_of`, ownership will act as renounced. For example NFT got burned or something happened with NFT contract.
+
+## Abstract Account Controlled Module
+
+For modules and contracts controlled by Abstract Accounts, we present a mechanism that allows those contracts to make sure that an in-coming message from the Account was originally called by an admin and not another module. This prevents any module to call admin functions on other modules and thus makes the module system more resistent to malicious modules.
+
+### Mechanism
+
+Modules and Account Owners can execute actions through the Account using the `account::ExecuteMsg::Execute` message variant. In order to execute an admin call, owners need to call `account::ExecuteMsg::AdminExecute`. This function will in order:
+
+- Set the `CALLING_TO_AS_ADMIN` storage item to the target of the admin call.
+- Call the specified function on the target module or contract.
+- Remove the `CALLING_TO_AS_ADMIN` storage item.
+
+In order to check that the call is an admin call, the target module or contract needs to check that the `CALLING_TO_AS_ADMIN` storage item is set on the account contract. If it's not set, it should error, as the call is not an authorized admin call.
+
+### Usage inside a module
+
+To use this functionality, Abstract provides helpers in form of the `NestedAdmin` structure. This structure should be used to handle `Abstract Accounts` as admin of a contract.
 
-### NFT governance type
+The `NestedAdmin::assert_admin` function will only return an `Result::Ok` if any of those conditions is true:
 
-In case NFT contract does not return owner of `owner_of`, ownership will act as renounced. For example NFT got burned or something happened with NFT contract.
+- The caller is the saved Account AND the `CALLING_TO_AS_ADMIN` variable is set on the account to either:
+  - The contract account address (`env.contract.address` is supposed to be fed to the self_addr variable)
```

Comment:
Added a Readme. Because of this line, I think we should take in `Env` instead of `self_addr` otherwise, it's simple to exchange the 2 addresses required.

Reply:
@CyberHoward wdyt ?

Reply:
Yeah agree. Realized it was a bad change when I was reviewing as well.

Reply:
`&Env` preferably